### ISSUE TITLE
Query option doesn't work on untype scenario

### ISSUE THREAD

**Issue by LianwMS** [OData/WebApi#175]
Suppose we have an untype Edm model as:  

```
<?xml version="1.0" encoding="utf-8"?>
<edmx:Edmx Version="4.0" xmlns:edmx="http://docs.oasis-open.org/odata/ns/edmx">
  <edmx:DataServices>
    <Schema Namespace="SNL" xmlns="http://docs.oasis-open.org/odata/ns/edm">
      <EntityType Name="Address">
        <Key>
          <PropertyRef Name="Id" />
        </Key>
        <Property Name="Id" Type="Edm.Int32" />
        <Property Name="Street" Type="Edm.String" />
        <Property Name="City" Type="Edm.String" />
      </EntityType>
      <EntityType Name="Server">
        <Key>
          <PropertyRef Name="KeyServers" />
        </Key>
        <Property Name="KeyServers" Type="Edm.String" />
        <Property Name="Server" Type="Edm.String" />
        <NavigationProperty Name="ServerAddresses" Type="Collection(SNL.Address)" ContainsTarget="true" />
      </EntityType>
      <EntityContainer Name="Container">
        <EntitySet Name="Servers" EntityType="SNL.Server">
          <NavigationPropertyBinding Path="ServerAddresses" Target="Addresses" />
        </EntitySet>
        <EntitySet Name="Addresses" EntityType="SNL.Address" />
      </EntityContainer>
    </Schema>
  </edmx:DataServices>
</edmx:Edmx>
```

When we create a controller and return the untype data as:  

```
public class ServersController : ODataController
    {
        [EnableQuery]
        public IHttpActionResult Get()
        {
EdmEntityObjectCollection servers =
                    new EdmEntityObjectCollection(
                        new EdmCollectionTypeReference(
                            new EdmCollectionType(new EdmEntityTypeReference(EdmModelBuilder.ServerEntityType, false))));

            int addressId = 1;
            for (int i = 0; i < 5; i++)
            {
                EdmEntityObject server = new EdmEntityObject(EdmModelBuilder.ServerEntityType);
                server.TrySetPropertyValue("KeyServers", i.ToString());
                server.TrySetPropertyValue("Server", "Test" + i);

                EdmEntityObjectCollection addresses =
                    new EdmEntityObjectCollection(
                        new EdmCollectionTypeReference(
                            new EdmCollectionType(new EdmEntityTypeReference(EdmModelBuilder.AddressEntityType, false))));

                for (int j = 0;  j < 3; j++)
                {
                    EdmEntityObject address = new EdmEntityObject(EdmModelBuilder.ServerEntityType);
                    address.TrySetPropertyValue("Id", addressId);
                    address.TrySetPropertyValue("Street", "Street #" + addressId);
                    address.TrySetPropertyValue("City", "City #" + addressId);
                    addressId++;

                    addresses.Add(address);
                }

                server.TrySetPropertyValue("ServerAddresses", addresses);
                servers.Add(server);
            }
            return Ok(servers);
        }
    }
}
```

We can query:  
*\* GET ~/odata/Servers *\*  
the payload:  

```
{
  "@odata.context":"http://localhost:40502/odata/$metadata#Servers","value":[
    {
      "KeyServers":"0","Server":"Test0"
    },{
      "KeyServers":"1","Server":"Test1"
    },{
      "KeyServers":"2","Server":"Test2"
    },{
      "KeyServers":"3","Server":"Test3"
    },{
      "KeyServers":"4","Server":"Test4"
    }
  ]
}
```

but, if we apply the query option, it doesn't work:  

_ GET ~/data/Servers?$expand=ServerAddresses  
GET ~/data/Servers?$select=Server  _  
## Work Item Details

**Original CodePlex Issue:**    [Issue 2204](http://aspnetwebstack.codeplex.com/workitem/2204)
**Status:** Proposed
**Reason Closed:**  Unassigned
**Assigned to:**    xuzhg
**Reported on:**    Dec 5, 2014 at 11:15 AM
**Reported by:**    xuzhg
**Updated on:** Dec 24, 2014 at 6:11 AM
**Updated by:** xuzhg

**Comment by LianwMS** [comment]
On *2014-12-11 17:49:13 UTC*, **cysu** commented:

Assign to Sam for an investigation.

**Comment by ntfirewall** [comment]
Are there any work around to this issue. I am trying to generate model based on user permissions. This seemed promising  WebApi/v4/DynamicEdmModelCreation, until realised odata queries do not work.

**Comment by habbes** [comment]
Adding this to the backlog to be reviewed in the next prioritization cycle. Backlog reference: 1791150

**Comment by habbes** [comment]
Closed as duplicate of #165
